lib/tasks/llm/index.js

```diff
@@ -36,6 +40,106 @@ class TaskLlm extends Task {
   async kill(cs) {
     super.kill(cs);
     await this.llm.kill(cs);
+    // clean up MCP clients
+    for (const mcpClient of this.mcpClients) {
+      const {client} = mcpClient;
+      if (client) {
+        await client.close();
+        this.logger.debug({url: mcpClient.url}, 'TaskLlm:kill - mcp client closed');
+      }
+    }
+    this.mcpClients = [];
+  }
+
+  async _getMcpClients() {
```

Comment:
I think we should factor this mcp client stuff into is own class under utils

Reply:
fixed